src/components/GameField/index.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { Component, RefObject } from 'react';
+import { Ball } from '../../types';
+import { findPathToTarget, COLORS } from '../../utils';
+import './index.scss';
+
+interface GameFieldProps {
+  fieldSize: number,
+  ballsCount: number,
+  gameFieldState: number[][],
+  playSound: boolean,
+  moveBallToNewCell: (ball: Ball, targetRow: number, targetColumn: number, path: [number, number][]) => void,
+}
+
+interface GameFieldState {
+  currentActive: Ball,
+  imagesLoaded: boolean,
+}
+
+export default class GameField extends Component<GameFieldProps, GameFieldState> {
+  timerId: NodeJS.Timeout;
+
+  audioRef: RefObject<HTMLAudioElement>;
+
+  imagesCache: HTMLImageElement[];
+
+  constructor(props: GameFieldProps) {
+    super(props);
+    this.state = {
+      currentActive: null,
+      imagesLoaded: false,
+    };
+    this.audioRef = React.createRef();
+    this.imagesCache = [];
+    const checkImage = (path: string, index: number) =>
+    new Promise((resolve, reject) => {
+        const img = new Image()
+        img.onload = () => {
+          this.imagesCache[index] = img;
+          resolve(`./img/${path}`);
+        }
+        img.onerror = () => reject()
+
+        img.src = `./img/${path}`;
+    })
+    Promise.all(
+    COLORS.map(checkImage)).then(() => this.setState({imagesLoaded: true}))
```

Comment:
You could create a method like "imageLoader" in ./utils and replace it from constructor.
As you wish.